File: web/src/domain/entities/HabitTrack.ts
/** Domain entity representing a 30-day habit track. */
export interface HabitTrack {
  id: string;
  goalId: string;
  userId: string;
  startDate: string;
  currentDay: number; // 1-30
  isFinished: boolean;
  createdAt: string;
  updatedAt: string;
}

export type CheckpointStatus = 'PENDING' | 'COMPLETED' | 'MISSED';

export interface HabitCheckpoint {
  id: string;
  habitTrackId: string;
  dayNumber: number; // 1-30
  description: string;
  status: CheckpointStatus;
  completedAt: string | null;
}

File: android/domain/src/main/kotlin/com/dopashift/domain/entity/HabitTrack.kt
package com.dopashift.domain.entity

import java.time.LocalDate
import java.util.UUID

/**
 * A 30-day sequence of micro-habit checkpoints tied to a GoalProfile.
 */
data class HabitTrack(
    val id: UUID,
    val goalId: UUID,
    val userId: UUID,
    val startDate: LocalDate,
    val currentDay: Int,
    val isFinished: Boolean = false,
    val checkpoints: List<HabitCheckpoint> = emptyList()
) {
    init {
        require(currentDay in 1..30) { "Current day must be between 1 and 30" }
    }
}
